api/src/test/java/org/apache/iceberg/expressions/TestEvaluator.java

```diff
@@ -624,47 +622,28 @@ public void testIn() {
 
   @Test
   public void testInExceptions() {
-    AssertHelpers.assertThrows(
-        "Throw exception if value is null",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> in("x", (Literal) null));
-
-    AssertHelpers.assertThrows(
-        "Throw exception if value is null",
-        NullPointerException.class,
-        "Values cannot be null for IN predicate",
-        () -> in("x", (Collection<?>) null));
-
-    AssertHelpers.assertThrows(
-        "Throw exception if calling literal() for IN predicate",
-        IllegalArgumentException.class,
-        "IN predicate cannot return a literal",
-        () -> in("x", 5, 6).literal());
-
-    AssertHelpers.assertThrows(
-        "Throw exception if any value in the input is null",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> in("x", 1, 2, null));
-
-    AssertHelpers.assertThrows(
-        "Throw exception if binding fails for any element in the set",
-        ValidationException.class,
-        "Invalid value for conversion to type int",
-        () -> new Evaluator(STRUCT, in("x", 7, 8, 9.1)));
-
-    AssertHelpers.assertThrows(
-        "Throw exception if no input value",
-        IllegalArgumentException.class,
-        "Cannot create IN predicate without a value",
-        () -> predicate(Expression.Operation.IN, "x"));
-
-    AssertHelpers.assertThrows(
-        "Implicit conversion IN to EQ and throw exception if binding fails",
-        ValidationException.class,
-        "Invalid value for conversion to type int",
-        () -> new Evaluator(STRUCT, predicate(Expression.Operation.IN, "x", 5.1)));
+    Assertions.assertThatThrownBy(() -> in("x", (Literal) null))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessageContaining("Cannot create expression literal from null");
```

Comment:
better to add a newline after each statement and would also be good to double-check whether we can use `.hasMessage()` here everywhere